src/utils/string_to_enum.C

```diff
@@ -54,25 +54,28 @@ namespace {
 
 
 // Reverse a map
-template <typename MapIter, class MapType>
+template <class MapType>
 inline
-void build_reverse_map (MapIter it, MapIter end, MapType & reverse)
+std::map<typename MapType::mapped_type, typename MapType::key_type>
+build_reverse_map (const MapType & forward)
 {
-  reverse.clear();
+  std::map<typename MapType::mapped_type, typename MapType::key_type> reverse;
```

Comment:
Could you just use `MapType`? Unless you really want to enforce it being a `map` vs. `unordered_map` (or maybe you want to avoid the possibility of multimap?)

Reply:
Ah, of course